deepstream/pravega-dev.Dockerfile

```diff
@@ -53,27 +53,39 @@ RUN set -eux; \
 
 WORKDIR /usr/src/gstreamer-pravega
 
+FROM builder-base as planner
+RUN cargo install cargo-chef
+COPY . .
+RUN cargo chef prepare  --recipe-path recipe.json
+
+FROM builder-base as cacher
+RUN cargo install cargo-chef
+COPY --from=planner /usr/src/gstreamer-pravega/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+
+FROM builder-base as final
 ## Build gst-plugin-pravega.
 
+# Copy over the cached dependencies
+COPY --from=cacher /usr/src/gstreamer-pravega/target target
+COPY --from=cacher /usr/local/cargo /usr/local/cargo
+COPY . .
```

Comment:
For the future, we should be sure to add everything possible to .dockerignore to avoid invalidating the cache unnecessarily.